compute/clusters.go

```diff
@@ -2,14 +2,17 @@ package compute
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"log"
+	"reflect"
 	"sort"
 	"strings"
 	"time"
 
 	"github.com/databrickslabs/databricks-terraform/common"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/ulule/deepcopier"
```

Comment:
why do we need external dependency? can we do without it?

Reply:
copying of the data using reflection is kind of black magic (for me), I've tried to implement it myself, but gave up after several tries - maybe if I spend more time, I could do it, but not sure that this is a best use of the time

Reply:
Frankly speaking, it works without this custom marshaling - it simply sends both `num_workers` & `autoscale` if both present, and control plane always prefer `autoscale`. I simply afraid that it may stop this way at some point of time, and we'll need to do hotfix release...

Reply:
Single node clusters aren’t major dbu bringers, so I’m okay with that risk. 

Additional libraries just require more processes and adding it just to use in one place might be an overkill :(

Reply:
My main concern around this marshalling - if API stop accept both `num_worker` & `autoscale`, this will affect everything, not just single node. If you're ok, I'll revert that commit.